src/eduid_common/authn/utils.py

```diff
@@ -151,6 +155,54 @@ def check_previous_identification(session_ns):
     return eppn
 
 
+def maybe_xml_to_string(message, logger=None):
+    """
+    Try to parse message as an XML string, and then return it pretty-printed.
+
+    If message couldn't be parsed, return string representation of it instead.
+
+    This is used to (debug-)log SAML requests/responses in a readable way.
+
+    :param message: XML string typically
+    :param logger: logging logger
+    :return: something ready for logging
+    :rtype: string
+    """
+    if isinstance(message, six.binary_type):
+        # message is returned as binary from pysaml2 in python3
+        message = message.decode('utf-8')
+    message = str(message)
+    try:
+        from defusedxml import ElementTree as DefusedElementTree
+        parser = DefusedElementTree.DefusedXMLParser()
+        xml = DefusedElementTree.XML(message, parser)
+        return DefusedElementTree.tostring(xml)
+    except Exception as exc:
+        if logger is not None:
+            logger.debug("Could not parse message of type {!r} as XML: {!r}".format(type(message), exc))
+        return message
+
+
+def init_pysaml2(cfgfile):
```

Comment:
This was a method in IdPApplication, moved here but not yet used in eduid-common - the idea is to use it to directly parse the saml request, either coming from an SP or from the session (instead of using a saml server attached to the cherrypy app). This would happen in the next PR I've mentioned before.